sdk/rust-examples/minigrep/src/main.rs

```diff
@@ -0,0 +1,53 @@
+//! Mini Grep
+//!
+//! ## Context
+//!
+//! Searches for the argument inside the input file, and prints the result.
+//!
+//! ## Authors
+//!
+//! The Veracruz Development Team.
+//!
+//! ## Copyright
+//!
+//! See the file `LICENSING.markdown` in the Veracruz root directory for licensing and
+//! copyright information.
+
+use anyhow::{self, Ok};
+use std::{env, fs};
+
+const INPUT_FILENAME: &'static str = "/input/random_search_text.txt";
+const OUTPUT_FILENAME: &'static str = "/output/search_results.txt";
+
+fn main() -> anyhow::Result<()> {
+    let args: Vec<String> = env::args().collect();
+    let query: String = args.get(0).unwrap().to_string(); // Argument: To be searched for
+
+    let file_vec: Vec<u8> = fs::read(INPUT_FILENAME)?;
+
+    let filename: String = String::from_utf8_lossy(&file_vec).to_string(); // Input file: To be searched in
+
+    let res_vec: Vec<&str> = search(query.as_str(), &filename.as_str());
```

Comment:
I think

```rust
query.as_str()
```

is more idiomatically written `&query` in Rust, and similarly

```rust
&filename.as_str()
```

can be simplified to `&filename`.